dashboard/app/static/coverage.js

```diff
@@ -11,6 +11,34 @@ function initTogglers(){
   });
 }
 
+function initUpdateForm(){
+  var curUrlParams = new URLSearchParams(window.location.search);
+  $('#target-period').val(curUrlParams.get('period'));
+  $('#target-subsystem').val(curUrlParams.get('subsystem'));
+  $('#target-manager').val(curUrlParams.get('manager'));
+  $("#only-unique").prop("checked", curUrlParams.get('subsystem') == "1");
+
+  $(document).ready(function() {
+    $("#updateButton").click(function(event) {
+      event.preventDefault(); // don't submit the form
+
+      var period = $("#target-period").val();
+      var subsystem = $("#target-subsystem").val();
+      var manager = $("#target-manager").val();
+      var onlyUnique = $("#only-unique").prop("checked") ? "1" : "0";
+
+      var newLocation = new URL(window.location.href);
```

Comment:
Why not just submit the form? If you set `name=` on the form elements and set `method="get"` on the form, these will be automatically set as arguments.

Reply:
Proposed approach forwards optional get parameters. I suspect `method="get"` will use only the form fields and drop existing parameters.
It was not my goal. But now it is a positive side-effect.

Reply:
https://syzkaller.appspot.com/upstream/graph/coverage_heatmap?period=month&period_count=1

`period_count` is a valid parameter, but it wan't mentioned on the form.
"Update" button doesn't touch it now and just forwards its value to backend.

Reply:
Hidden form field is the alternative here. `method="get"` may give shorter code.

Reply:
Let me check how will it look like.

Reply:
A hidden field sounds better than hacking `searchParams`.

By the way, how could `period_count` have become 1? It's not set anywhere.

```
$ git grep -r "period_count"
dashboard/app/graphs.go:        periodCount := r.FormValue("period_count")
```

If it's something important, maybe it's worth also including into the non-hidden part of the form?

Reply:
Thanks for `method="get"`. Let's see how will it work.

Reply:
I like it. It is much shorter.

Reply:
I'll add the `period_count` control separately. Hidden form values looks better for the optional arguments.